config/config.go

```diff
@@ -1272,7 +1272,11 @@ type StorageConfig struct {
 	DiscardABCIResponses bool `mapstructure:"discard_abci_responses"`
 	// Configuration related to storage pruning.
 	Pruning *PruningConfig `mapstructure:"pruning"`
-
+	// Compaction config
+	CompactOnPruning bool `mapstructure:"compact_on_pruning"`
```

Comment:
```suggestion
	Compact bool `mapstructure:"compact"`
```

Do we have to tie compaction to pruning?

Reply:
I did experiments and if we do not prune, compaction makes no sense. It makes sense to do it only if we actually delete stuff.

Reply:
What if we take into account https://github.com/cometbft/cometbft/pull/2017?

Reply:
I see.. We could then rename compact to be just what you are suggesting but in that context, should CompactionInterval be applied to the ABCI results the same way as it is for blocks or would you call the `db.Compact` function from #2017 every time you delete? Note that in that case, if the pruner is pruning and you prune here as well we might trigger compaction more frequently on the state store.

Reply:
What's the overhead of calling pruning every block?

Reply:
Depends on the size of the DB. If pruning is enabled and state is pruned frequently and the DB is not large it is not that big. But if you prune rarely, or not at all but you just want to compact the DB because you want to delete these ABCI results then it will traverse all the levels of the DB on every call.

Reply:
> should CompactionInterval be applied to the ABCI results the same way as it is for blocks or would you call the db.Compact function from https://github.com/cometbft/cometbft/pull/2017 every time you delete?

Then I would say the former (i.e. the same way it is for blocks).

Reply:
Do you think it would then make sense, once this is merged to add a call to compact ABCI results from #2017? I am wondering whether we should then not call `Compact` within the `PruneStates` and `PruneBlocks` function but rather extend the interfaces of the state and block store to include a `Compact` function. Then in the pruner service , if the retain heights changed call Compact. If they have not changed but 

``` 
if config.Compact && discardABCIResults = false  && height % CompactionInterval == 0  {

   stateStore.Compact(nil,nil)
   
}
```

Before making this change we just need to make sure this is not overkill for achieving the effect in 2017? If it makes more sense from a design perspective then I am happy to do this change.

Reply:
> Then in the pruner service , if the retain heights changed call Compact. If they have not changed but

This makes sense, but let's delay this decision until #2017.

I'm in favor of renaming `CompactOnPruning` to just `Compact`, though.

Reply:
Ok yes I am totally fine with this. Then lets try and merge this logic as is but make sure we finalize the changes before v1 as they are potentially breaking afterwards.  I'll edit the name of the config variable. I realized there is a potential corner case where the data companion is enabled but the state and block stores are never pruned because the app does not allow it. The data companion can still pruned the ABCI results and indexer and we should explicitly call compact in those cases (not relying that state and block stores will be pruned). Will push that change in a min